src/routers/datatypeRouters/edges/variants_genes.ts

```diff
@@ -94,76 +71,21 @@ async function conditionalSearch (input: paramsFormatType, type: string): Promis
     delete input.gene_id
   }
 
-  if (type === 'eqtl') {
-    input.label = 'eQTL'
-    return await routerEqtls.getEdgeObjects(input, '', verbose)
-  }
-
-  const preProcessed = preProcessRegionParam({ ...input, ...{ sort: '_key' } }, null, 'intron')
-
-  if (type === 'sqtl') {
-    input.label = 'splice_QTL'
-    return await routerSqtls.getEdgeObjects(preProcessed, '', verbose)
-  }
-
-  return await routerQtls.getEdgeObjects(preProcessed, '', verbose)
-}
-
-// variant ID -(qtls)-> genes, genes -> transcripts, transcripts -> proteins
-async function asb (variantId: string, verbose: boolean): Promise<asbType> {
-  const variantGenes = await routerQtls.getTargetSet(['variants/' + variantId])
-
-  if (variantGenes.length === 0) {
-    throw new TRPCError({
-      code: 'NOT_FOUND',
-      message: `Variant ${variantId} not found.`
-    })
-  }
-
-  const geneTranscripts = await routerGenesTranscripts.getTargetSet(variantGenes[0].genes)
-
-  for (let i = 0; i < geneTranscripts.length; i++) {
-    geneTranscripts[i].transcripts = await routerTranscriptsProteins.getTargetSet(geneTranscripts[i].transcripts, verbose)
-  }
-
-  variantGenes[0].genes = geneTranscripts
-  return variantGenes[0]
+  return await routerQtls.getEdgeObjects(input, '', verbose)
 }
-
-const sqtlFromVariants = publicProcedure
-  .meta({ openapi: { method: 'GET', path: '/variants/s-qtls' } })
-  .input(variantsSqtlQueryFormat)
-  .output(z.array(sqtlFormat))
-  .query(async ({ input }) => await conditionalSearch(input, 'sqtl'))
-
-const eqtlFromVariants = publicProcedure
-  .meta({ openapi: { method: 'GET', path: '/variants/e-qtls' } })
-  .input(variantsEqtlQueryFormat)
-  .output(z.array(eqtlFormat))
-  .query(async ({ input }) => await conditionalSearch(input, 'eqtl'))
-
 const genesFromVariants = publicProcedure
-  .meta({ openapi: { method: 'GET', path: '/variants/{variant_id}/genes' } })
-  .input(z.object({ variant_id: z.string() }).merge(variantsEqtlQueryFormat))
+  .meta({ openapi: { method: 'GET', path: '/variants/genes' } })
+  .input(z.object({ variant_id: z.string().optional() }).merge(variantsQtlsQueryFormat))
   .output(z.array(eqtlFormat.merge(sqtlFormat)))
-  .query(async ({ input }) => await conditionalSearch(input, 'all'))
+  .query(async ({ input }) => await qtlSearch(input))
 
 const variantsFromGenes = publicProcedure
   .meta({ openapi: { method: 'GET', path: '/genes/{gene_id}/variants' } })
```

Comment:
`path: '/genes/{gene_id}/variants' ` should be `path: '/variants/genes' `?

Reply:
It should be /genes/variants, this was addressed in the genes PR